segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperLog.java

```diff
@@ -590,6 +648,9 @@ private void addCallback(Long entryId, Throwable error, Write write) {
         try {
             if (error == null) {
                 assert entryId != null;
+                // Expected and actual entryIds should always match.
+                Preconditions.checkState(entryId == write.getExpectedEntryId(), String.format("Wrong expected entry id (%d) compared to actual one (%d) for ledger %d",
```

Comment:
if "expectedentryid" we maintain goes out of sync with actual entry id for some reason, it would lead to a crash, followed by a rollover leading to resetting of "expectedentryid"   is it?

Reply:
I have changed the approach, now is simple and we basically keep track of the "confirmed" writes (i.e., the ones that have executed the `addCallback()`). If we handle a closed ledger and we realize that the LAC is higher than the last confirmed write, then we just complete the queued writes up to LAC. I think that with this approach we do not really care about getting out of sync due to rollovers.